_posts/uportal/2017-12-06-myuw-fork.md

```diff
@@ -0,0 +1,168 @@
++ Add `/marketplace/entry/{fname}.json` to get JSON for a single Marketplace entry
++ Add `/portlet/{fname}.json` to get JSON for a single portlet
++ Add `GroupRESTController`
++ Defaulting `url` of LayoutPortlet to that to render the portlet via uPortal
++ Add `lifecycleState` to `LayoutPortlet`
++ Add simplified move-portlet API call (`action=movePortletAjax` in `UpdatePreferencesServlet`)
++ Add remove-from-layout-by-fname (`action=removeByFName` in `UpdatePreferencesServlet`)
++ Add `firstName`, `lastName` in the `SessionRESTController` response. Also, `originalUsername`  when swapped.
++ Add some error handling in `LayoutPortlet` to catch when `widgetConfig` is not valid JSON.
++ Increase time-to-live of `marketplacePortletDefinitionCache`
++ Filter marketplace entries to only those in `PUBLISHED` publication lifecycle state. (Arguably this should be more nuanced, reflecting the viewing user's permissions -- maybe you've got permission to see entries that are in states other than `PUBLISHED`).
++ Attempts to make hitting JSON APIs have the side effect of logging the user in / bootstrapping a uPortal session when they can. This is fraught.
+
+# Hacked DLM
+
+MyUW has modified DLM to simplify the user experience of adding content to one's homepage.
+
++ Default adding a new item to the first rather than last position in a container.
++ Reset user layout on all profiles rather than on just the currently selected profile. (The feature but maybe not the configuration to use it is in upstream).
++ Various low value hacks.
+
+# Hacked login process
+
+MyUW has modified uPortal login to try to improve the experience for users who 
+navigate directly to `uPortal-home` with only some or none of the uPortal 
+session they need already established.
+
++ Added "silent login" feature to `LoginController` to enable `uPortal-home` 
+bootstrapping a session via async HTTP request
++ Apply `requestAttributeSourceFilter` to API requests as well as to `/Login`
+
+# Removed features
+
+MyUW has removed features that MyUW doesn't seem to need.
+
++ Removed "related apps". Big performance improvement.
++ Set 
+`org.jasig.portal.portlets.marketplacePortlet.loadMarketplaceOnLogin=false` . 
+Stability improvement.
++ Removed `x:SearchRequest` and `x:SearchResponse` event handling in 
+`SimpleContentPortlet` `portlet.xml`
++ Removed jGroups cache invalidation signaling and listening
++ In process of removing Categories, use just one giant category that conveys 
+no permissions, assign all SUBSCRIBE and BROWSE permissions on a 
+per-portlet-definition basis
+
+# Other fixes
+
++ Fix deleting permission grants via permissions management Portlet
```

Comment:
Maybe rephrase? This could be ambiguous for folks that don't have in depth knowledge of portals permission system.